src/market/directory.rs

```diff
@@ -30,7 +30,9 @@ use tokio::{
 /// Represents errors that can occur during directory server operations.
 #[derive(Debug)]
 pub enum DirectoryServerError {
-    Other(&'static str),
+    PoissonError(&'static str),
```

Comment:
For Poison error, we can follow the pattern used in MakerError here https://github.com/citadel-tech/coinswap/blob/73e17a80ac10d42b97966430d745d1bf90155563/src/maker/error.rs#L43-L53

No need for static str inside, and will also remove the map_err at call site.

Reply:
Agreed.